lib/gollum/views/pages.rb

```diff
@@ -41,19 +43,23 @@ def files_folders
               folder_path = @path ? "#{@path}/#{folder}" : folder
               folder_link = %{<li><a href="#{@base_url}/pages/#{folder_path}/" class="folder">#{folder}</a></li>}
 
-              unless folder_links.include?(folder_link)
-                folder_links << folder_link
-
-                folder_link
-              end
+              folders[folder] = folder_link unless folders.key?(folder_link)
```

Comment:
Ok, so I think I changed that at the last second before the commit and mixed it up. I think it makes more sense for it to be:

```ruby
folders[folder] = folder_link unless folders.key?(folder)
```

That way, each `folder` is only rendered once. I think I would have caught this with a more thorough test case (which I intend on writing)